### PR TITLE
added #79 chat component

### PR DESCRIPTION
**Pull request by rohitgupta274509**
please review and approve it

### CODE REVIEW

**Review by ankitvats1 — APPROVED** [review]

**Review by Rahul139-stack — APPROVED** [review]

File: src/components/Chat/ChatComponent.vue [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+      <br :class="send">
+      {{ message.body }}
+        </p>
+    
+        <div class="iframe-container">
+             <form @submit.prevent="sendMessage('in')" id="sender-form" class="message-inputtab">
+                <div class="p-inputgroup">
+                 <InputText v-model="Message" placeholder= "Type your message here..." id="sender-input" class="chat-inputs" />
+                     <Button label="send" class="p-button-primary" type=" submit">
+                         <div class="sendbutton">Send</div></Button>
```
Comment (wildone) [inline src/components/Chat/ChatComponent.vue:25]: is this i18n?

File: src/components/Chat/ChatComponent.vue [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    <div class="header" :class="{ 'menu-hidden': !mainMenuVisible }">
+      <main-menu
+        :main-menu-visible="mainMenuVisible"
+        @toggle="toggleMainMenu"
+      />
+      <menu-bar :menu-bar-visible="mainMenuVisible" @toggle="toggleMainMenu" />
+    </div>
+    <!-- content -->
+    <div class="chatpage">
+        <p v-for="message in messages" :key="message.id" class="message" :class="{ 'message-out': message.author === 'Receiver', 'message-in': message.author !== 'Receiver' }">
```
Comment (wildone) [inline src/components/Chat/ChatComponent.vue:12]: this should relly on message type.

File: src/components/Chat/ChatComponent.vue [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        {{message.author}}
+      </span>
+      <br :class="send">
+      {{ message.body }}
+        </p>
+    
+        <div class="iframe-container">
+             <form @submit.prevent="sendMessage('in')" id="sender-form" class="message-inputtab">
+                <div class="p-inputgroup">
+                 <InputText v-model="Message" placeholder= "Type your message here..." id="sender-input" class="chat-inputs" />
```
Comment (wildone) [inline src/components/Chat/ChatComponent.vue:23]: can we make this i18n?

File: src/components/Chat/ChatComponent.vue [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+  export default {
+    name: "Maps",
+    components: { MenuBar, MainMenu, InputText, Button },
+    data() {
+      return {
+        showMainOverlayMenu: false,
+         mainMenuVisible: true,
+         Message: '',
+         messages: 
+            [
```
Comment (wildone) [inline src/components/Chat/ChatComponent.vue:47]: can we add this to Chat tab in home?

File: src/views/Home.vue [hunk 5]
```diff
@@ -1,16 +1,18 @@
...
   <charts v-else-if="$route.params.id === 'charts'" />
+  <chat v-else-if="$route.params.id === 'chat'" />
   <project v-else />
 </template>
 
 <script>
   import Project from "@/components/Project"
   import Charts from "@/components/Charts"
   import Maps from "@/components/Maps"
+  import Chat from "@/components/Chat/ChatComponent.vue"
```
Comment (wildone) [inline src/views/Home.vue:12]: can we rename Chat/ChatCompoent.vue to Chat/index.vue and use it like folder above?

File: src/components/MenuBar.vue [hunk 6]
```diff
@@ -31,7 +31,7 @@
       <i
         v-if="theme === 'light'"
         class="pi pi-sun"
-        style="font-weight: bold"
+        style="font-weight: bold;"
```
Comment (wildone) [inline src/components/MenuBar.vue:34]: can we move all style to scss file?

File: src/components/Chat/ChatComponent.vue [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+  </div>
+</template>
+
+<script>
+  import MenuBar from "@/components/MenuBar.vue"
+  import MainMenu from "@/components/MainMenu.vue"
+  import InputText from "primevue/inputtext"
+  import Button from "primevue/button"
+  export default {
+    name: "Maps",
```
Comment (wildone) [inline src/components/Chat/ChatComponent.vue:39]: whats this for?

File: src/components/Chat/ChatComponent.vue [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+    methods: {
+      toggleMainMenu() {
+        this.mainMenuVisible = !this.mainMenuVisible
+      },
+      sendMessage(direction) {
+      if (!this.Message && !this.Message) {
+        return
+      }
+       if (direction === 'in') {
+        this.messages.push({body: this.Message, author: 'Sender'})
```
Comment (wildone) [inline src/components/Chat/ChatComponent.vue:68]: maybe we need more data

- timestamp
- body
- direction (to/from)
- author-to
- author-from
- type (message)
- read
- delivered
- id

File: src/components/Chat/ChatComponent.vue [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+        this.SenderMessage = ''
+        this.messages.push({body: this.Message, author: 'Receiver',})
+        this.Message = ''
+      } else {
+        alert('something went wrong')
```
Comment (wildone) [inline src/components/Chat/ChatComponent.vue:73]: should this be in i18n? can we raise a new ticket for creating new error store please.  Leave a //todo: raise error event

**Review by wildone — CHANGES_REQUESTED** [review]

File: src/components/Chat/ChatComponent.vue (hunk 1 above)
Comment (rohitgupta274509) [reply src/components/Chat/ChatComponent.vue]: added i18n.

File: src/components/Chat/ChatComponent.vue (hunk 2 above)
Comment (rohitgupta274509) [reply src/components/Chat/ChatComponent.vue]: updated  keywords

File: src/components/Chat/ChatComponent.vue (hunk 3 above)
Comment (rohitgupta274509) [reply src/components/Chat/ChatComponent.vue]: added i18n

File: src/components/Chat/ChatComponent.vue (hunk 4 above)
Comment (rohitgupta274509) [reply src/components/Chat/ChatComponent.vue]: deleted this part, not required here.

File: src/views/Home.vue (hunk 5 above)
Comment (rohitgupta274509) [reply src/views/Home.vue]: Rename ChatCompoent.vue to index.vue

File: src/components/MenuBar.vue (hunk 6 above)
Comment (rohitgupta274509) [reply src/components/MenuBar.vue]: added in scss file

File: src/components/Chat/ChatComponent.vue (hunk 7 above)
Comment (rohitgupta274509) [reply src/components/Chat/ChatComponent.vue]: updated name as Chat

File: src/components/Chat/ChatComponent.vue (hunk 8 above)
Comment (rohitgupta274509) [reply src/components/Chat/ChatComponent.vue]: added attributes like timestamp and delivered notation

File: src/components/Chat/ChatComponent.vue (hunk 9 above)
Comment (rohitgupta274509) [reply src/components/Chat/ChatComponent.vue]: added i18n .
new ticket for creating new error store - //todo

**Review by ankitvats1 — APPROVED** [review]

**Review by Shivam-apna — APPROVED** [review]

**Review by wildone — APPROVED** [review]

**Pull request merged**